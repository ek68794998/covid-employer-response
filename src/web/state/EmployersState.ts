import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}