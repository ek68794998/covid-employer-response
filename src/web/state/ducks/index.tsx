import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});