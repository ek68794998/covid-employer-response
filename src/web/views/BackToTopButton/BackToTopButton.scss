@import "../Colors.scss";
@import "../Dimensions.scss";

$top-button-size: 48px;
$top-button-size-small: 32px;

.BackToTopButton__Link {
	background: $background;
	border: 0;
	border-radius: ($top-button-size / 2);
	bottom: ($page-padding / 2);
	box-shadow: 0 2px 6px $box-shadow-color;
	cursor: pointer;
	position: fixed;
	height: $top-button-size;
	outline: none;
	pointer-events: initial;
	right: ($page-padding / 2);
	width: $top-button-size;
	z-index: 20;

	i {
		font-size: 36px;
		margin-top: 2px;
	}
}

@media (max-width: $body-narrow-width) {
	.BackToTopButton__Link {
		bottom: ($page-padding-narrow / 2);
		height: $top-button-size-small;
		right: ($page-padding-narrow / 2);
		width: $top-button-size-small;

		i {
			font-size: 24px;
			margin-left: -2px;
		}
	}
}
