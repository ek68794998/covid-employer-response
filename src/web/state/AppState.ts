import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}