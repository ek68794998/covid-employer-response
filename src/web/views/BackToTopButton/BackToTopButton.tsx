import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import { withRouter } from "react-router-dom";

import { LocalizedStrings } from "../../../common/LocalizedStrings";

import { getStrings } from "../../state/ducks/localization/selectors";

import "./BackToTopButton.scss";

const getIsVisible = (): boolean => typeof window === "undefined" ? false : window.scrollY > 0;

const BackToTopButton: React.FC = (): React.ReactElement | null => {
	const [ isVisible, setIsVisible ] = useState(getIsVisible());
	const strings: LocalizedStrings = useSelector(getStrings);

	if (typeof window !== "undefined") {
		const trackScrolling = (): void => {
			setIsVisible(getIsVisible());
		};

		useEffect(
			(): void => {
				trackScrolling();

				window.addEventListener("scroll", trackScrolling);

				return (): void => {
					window.removeEventListener("scroll", trackScrolling);
				};
			},
			[]);
	}

	if (!isVisible) {
		return null;
	}

	return (
		<button
			className="BackToTopButton__Link"
			onClick={(): void => { window.scrollTo(window.scrollX, 0); }}
			title={strings.backToTop}
		>
			<i className="material-icons">arrow_upward</i>
		</button>
	);
};

export default withRouter(BackToTopButton);
