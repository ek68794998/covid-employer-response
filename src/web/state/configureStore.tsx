import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;