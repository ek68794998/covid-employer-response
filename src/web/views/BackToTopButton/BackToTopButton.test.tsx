import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { getPlocStringsAsync } from "../../../__tests__/TestUtils";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import BackToTopButton from "./BackToTopButton";

describe("<BackToTopButton />", () => {
	test("renders without exploding", async () => {
		const store: Store<AppState, AnyAction> = configureStore({ strings: await getPlocStringsAsync() });

		(global as any).scrollY = 0;

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<BackToTopButton />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});

	test("shows button when scrolled down", async () => {
		const store: Store<AppState, AnyAction> = configureStore({ strings: await getPlocStringsAsync() });

		(global as any).scrollY = 50;

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<BackToTopButton />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
