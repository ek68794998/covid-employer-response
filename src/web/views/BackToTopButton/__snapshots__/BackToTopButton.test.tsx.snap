// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackToTopButton /> renders without exploding 1`] = `null`;

exports[`<BackToTopButton /> shows button when scrolled down 1`] = `
<button
  className="BackToTopButton__Link"
  onClick={[Function]}
  title="βáçƙTôTôƥ"
>
  <i
    className="material-icons"
  >
    arrow_upward
  </i>
</button>
`;
